src/ansys/simai/core/data/model_configuration.py

```diff
@@ -330,6 +327,76 @@ def __get_build_preset(self):
 
     build_preset = property(__get_build_preset, __set_build_preset)
 
+    def __validate_variables(self, vars_to_validate: list[str], var_type: str):
+        sample_metadata = self.project.sample.fields.get("extracted_metadata")
+        var_fields = dict_get(sample_metadata, var_type, "fields", default=[])
+        var_fields_name = {fd.get("name") for fd in var_fields}
+        unknown_variables = set(vars_to_validate) - var_fields_name
+        if unknown_variables:
+            raise ProcessingError(
+                f"{self.__class__.__name__}: {var_type} variables {', '.join(unknown_variables)} do not exist in the reference sample."
+            ) from None
```

Comment:
AFAIK `from None` is only useful in a context where an exception could already have been thrown (in the `except` of a `try`)
https://docs.python.org/3/tutorial/errors.html#exception-chaining
I don't think this is the care here.

Reply:
exception chaining removed